packages/app/src/cli/services/deploy.ts

```diff
@@ -96,15 +101,25 @@ export async function deploy(options: DeployOptions) {
         {
           title: uploadTaskTitle,
           task: async () => {
-            const appModules = await Promise.all(
-              app.allExtensions.flatMap((ext) => ext.bundleConfig({identifiers, developerPlatformClient, apiKey})),
+            const appModules: BundleConfigWithSpecificationIdentifier[] = getArrayRejectingUndefined(
+              await Promise.all(
+                app.allExtensions.flatMap(async (ext): Promise<BundleConfigWithSpecificationIdentifier | undefined> => {
+                  const bundleConfig = await ext.bundleConfig({identifiers, developerPlatformClient, apiKey})
+                  if (bundleConfig) {
+                    return {
+                      ...bundleConfig,
+                      specificationIdentifier: developerPlatformClient.toExtensionGraphQLType(ext.graphQLType),
+                    }
```

Comment:
Why doesn't the `bundleConfig` function directly include this `specificationIdentifier` property instead of adding it here?